pkg/manifests/manifests_test.go

```diff
@@ -1599,6 +1601,131 @@ alertmanagerMain:
 	}
 }
 
+func TestK8sPrometheusAdapterAuditLog(t *testing.T) {
+	// setting audit level should add the audit args
+	tt := []struct {
+		config       string
+		expectedArgs []string
+		err          error
+	}{{
+		config: ``,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `k8sPrometheusAdapter: `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit: {} `,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/metadata-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: Request
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/request-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+k8sPrometheusAdapter:
+  audit:
+    profile: RequestResponse
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/requestresponse-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+     profile: None
+`,
+		expectedArgs: []string{
+			"--audit-policy-file=/etc/audit/none-profile.yaml",
+			"--audit-log-path=/var/log/adapter/audit.log",
+			"--audit-log-maxsize=100",
+		},
+		err: nil,
+	}, {
+		config: `
+  k8sPrometheusAdapter:
+    audit:
+      profile: Foobar  # should generate an error
+`,
+		err: ErrConfigValidation,
+	}}
+
+	for _, test := range tt {
+		c, err := NewConfigFromString(test.config)
+		if err != nil {
+			t.Logf("%s\n\n", test.config)
```

Comment:
Instead of logging the whole config for each failure, could you give a name to each test case and run them in subtests with `t.Run()`? This would already give us enough insights when debugging test failures.

Reply:
I am not sure I agree ... e.g  below provides a more readable test output 
```
=== RUN   TestK8sPrometheusAdapterAuditLog/None
    manifests_test.go:1686:
          k8sPrometheusAdapter:
            audit:
             profile: Noxne


    manifests_test.go:1687: Expected error %!q(<nil>) but got "invalid value for config - adapter audit profile: Noxne"
```
Than
```
=== RUN   TestK8sPrometheusAdapterAuditLog/None
    manifests_test.go:1706: k8s-prometheus-adapter audit is not configured correctly
        got:
                []string{"--audit-policy-file=/etc/audit/metadata-profile.yaml", "--audit-log-path=/var/log/adapter/audit.log", "--audit-log-maxsize=100"}

        expected:
                []string{"--audit-policy-file=/etc/audit/none-profile.yaml", "--audit-log-path=/var/log/adapter/audit.log", "--audit-log-maxsize=100"}
```
However, I have changed it